src/vlcPlayer.c

```diff
@@ -125,6 +123,10 @@ void open_media(GSList *list, int n_tracks, gboolean add)
    g_slist_free(list) ;
 
    initialize_gtk_playlist() ;
+   if (vlc->tracks[0]->type == AUDIO) {
```

Comment:
If you do it here, during track transitions it will be (not appreciably) faster. But when you have listened to a music playlist and then opening a film music icon remains and you have to double-click playlist button to be able to watch the video.

Reply:
thanks, wrote a new function `format_display_for_media ()`  to do this